File: Agents/Azure-Agent-Skills-In-Action/skill-demos/sdk-java/02_AzureIdentity.java
// TRIPLE:
//   Skill: azure-identity-java
//   Prompt: "Using azure-identity-java skill, write Java code that gets tokens for 3 scopes (ai.azure.com, cognitiveservices, graph)."
//   Deliverable: This file
//   Source: https://github.com/microsoft/skills/tree/main/.github/plugins/azure-sdk-java/skills/azure-identity-java

import com.azure.core.credential.AccessToken;
import com.azure.core.credential.TokenRequestContext;
import com.azure.identity.DefaultAzureCredential;
import com.azure.identity.DefaultAzureCredentialBuilder;

public class AzureIdentity {

    public static void main(String[] args) {
        DefaultAzureCredential credential = new DefaultAzureCredentialBuilder().build();

        String[] scopes = {
                "https://ai.azure.com/.default",
                "https://cognitiveservices.azure.com/.default",
                "https://graph.microsoft.com/.default"
        };

        for (String scope : scopes) {
            try {
                TokenRequestContext context = new TokenRequestContext().addScopes(scope);
                AccessToken token = credential.getTokenSync(context);
                System.out.printf("Scope: %s%n", scope);
                System.out.printf("  Token (first 20 chars): %s...%n",
                        token.getToken().substring(0, 20));
                System.out.printf("  Expires: %s%n", token.getExpiresAt());
            } catch (Exception e) {
                System.err.printf("Failed for scope %s: %s%n", scope, e.getMessage());
            }
        }
    }
}

File: Agents/Azure-Agent-Skills-In-Action/skill-demos/sdk-typescript/02_azure_identity.ts
// TRIPLE:
//   Skill: azure-identity-ts
//   Prompt: "Using azure-identity-ts skill, write TypeScript code that gets tokens for 3 scopes (ai.azure.com, cognitiveservices, graph)."
//   Deliverable: This file
//   Source: https://github.com/microsoft/skills/tree/main/.github/plugins/azure-sdk-typescript/skills/azure-identity-ts

import { DefaultAzureCredential } from "@azure/identity";

async function main(): Promise<void> {
  const credential = new DefaultAzureCredential();

  const scopes = [
    "https://ai.azure.com/.default",
    "https://cognitiveservices.azure.com/.default",
    "https://graph.microsoft.com/.default",
  ];

  for (const scope of scopes) {
    try {
      const token = await credential.getToken(scope);
      const expiresOn = token.expiresOnTimestamp
        ? new Date(token.expiresOnTimestamp).toISOString()
        : "N/A";
      console.log(`Scope: ${scope}`);
      console.log(`  Token (first 20 chars): ${token.token.substring(0, 20)}...`);
      console.log(`  Expires: ${expiresOn}`);
    } catch (err: any) {
      console.error(`Failed for scope ${scope}: ${err.message}`);
    }
  }
}

main().catch(console.error);
